src/wrapper.jl

```diff
@@ -1,47 +1,165 @@
-# LibFTD2XX.jl
+# LibFTD2XX.jl - C Library Wrapper
+#
+# By Reuben Hill 2019, Gowerlabs Ltd, reuben@gowerlabs.co.uk
+#
+# Copyright (c) Gowerlabs Ltd.
+#
+# This module contains wrappers for D2XX devices. See D2XX Programmer's Guide 
+# (FT_000071) for more information.
+#
+# Only recommended for advanced users. Note that there is minimal argument 
+# checking in these wrapper methods. 
 
-export FTWordLength, BITS_8, BITS_7,
-       FTStopBits, STOP_BITS_1, STOP_BITS_2,
-       FTParity, PARITY_NONE, PARITY_ODD, PARITY_EVEN, PARITY_MARK, PARITY_SPACE,
-       FTOpenBy, OPEN_BY_SERIAL_NUMBER, OPEN_BY_DESCRIPTION, OPEN_BY_LOCATION
+module Wrapper
 
-# Constants
+# Type Aliases
+export DWORD, ULONG, UCHAR
+
+# Library Constants
+export FT_OPEN_BY_SERIAL_NUMBER, FT_OPEN_BY_DESCRIPTION, FT_OPEN_BY_LOCATION
+export FT_DEVICE
+export FT_LIST_NUMBER_ONLY, FT_LIST_BY_INDEX, FT_LIST_ALL
+export FT_BITS_8, FT_BITS_7
+export FT_STOP_BITS_1, FT_STOP_BITS_2
+export FT_PARITY_NONE, FT_PARITY_ODD, FT_PARITY_EVEN, FT_PARITY_MARK, FT_PARITY_SPACE
+# not yet implemented...
+# export FT_FLOW_NONE, FT_FLOW_RTS_CTS, FT_FLOW_DTR_DSR, FT_FLOW_XON_XOFF
+# export FT_EVENT_RXCHAR, FT_EVENT_MODEM_STATUS, FT_EVENT_LINE_STATUS not yet implemented
+export FT_PURGE_RX, FT_PURGE_TX
+export FT_STATUS_ENUM,
+        FT_OK,
+        FT_INVALID_HANDLE,
+        FT_DEVICE_NOT_FOUND,
+        FT_DEVICE_NOT_OPENED,
+        FT_IO_ERROR,
+        FT_INSUFFICIENT_RESOURCES,
+        FT_INVALID_PARAMETER,
+        FT_INVALID_BAUD_RATE,
+        FT_DEVICE_NOT_OPENED_FOR_ERASE,
+        FT_DEVICE_NOT_OPENED_FOR_WRITE,
+        FT_FAILED_TO_WRITE_DEVICE,
+        FT_EEPROM_READ_FAILED,
+        FT_EEPROM_WRITE_FAILED,
+        FT_EEPROM_ERASE_FAILED,
+        FT_EEPROM_NOT_PRESENT,
+        FT_EEPROM_NOT_PROGRAMMED,
+        FT_INVALID_ARGS,
+        FT_NOT_SUPPORTED,
+        FT_OTHER_ERROR,
+        FT_DEVICE_LIST_NOT_READY
+
+# Types
+export FT_HANDLE
+
+# Functions
+export ptr
```

Comment:
Should this be exported? Quite likely candidate for user variable names/other packages

Reply:
This is not a case of[ type piracy](https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-type-piracy-1) - what's the issue if it's exported? The `Wrapper` submodule is only accessible as a submodule of `LibFTD2XX` anyway. You'd have to explicitly import it to screw up other people's namespaces

Reply:
resolved in e632c86bf430902c45b7c313c54c6f7ae40505b7